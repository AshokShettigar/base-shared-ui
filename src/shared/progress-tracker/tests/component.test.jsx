import React from 'react';
import { shallow } from 'enzyme';

import ProgressTracker from '../component';

const defaultProps = {
    activeStep: 1,
    steps: [{ title: 'Select targets' }, { title: 'Configure setup' }, { title: 'Schedule' }],
    className: 'className',
    stepsClassName: 'stepsClassName',
    stepClassName: 'stepClassName',
    stepIconClassName: 'stepIconClassName',
    stepTitleClassName: 'stepTitleClassName'
};

describe('Progress Tracker component', () => {
    it('should render component', () => {
        const wrapper = shallow(<ProgressTracker {...defaultProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with custom active step', () => {
        const wrapper = shallow(<ProgressTracker {...defaultProps} activeStep="2" />);

        expect(wrapper).toMatchSnapshot();
    });
});
