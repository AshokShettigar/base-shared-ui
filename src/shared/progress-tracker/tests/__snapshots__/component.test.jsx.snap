// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Tracker component should render component 1`] = `
<div
  className="root className"
  data-component="ProgressTracker"
>
  <ul
    className="steps stepsClassName"
  >
    <li
      className="step stepClassName current"
    >
      <i
        className="stepIcon stepIconClassName"
      >
        1
      </i>
      <span
        className="stepTitle stepTitleClassName"
      >
        Select targets
      </span>
    </li>
    <li
      className="step stepClassName"
    >
      <i
        className="stepIcon stepIconClassName"
      >
        2
      </i>
      <span
        className="stepTitle stepTitleClassName"
      >
        Configure setup
      </span>
    </li>
    <li
      className="step stepClassName"
    >
      <i
        className="stepIcon stepIconClassName"
      >
        3
      </i>
      <span
        className="stepTitle stepTitleClassName"
      >
        Schedule
      </span>
    </li>
  </ul>
</div>
`;

exports[`Progress Tracker component should render component with custom active step 1`] = `
<div
  className="root className"
  data-component="ProgressTracker"
>
  <ul
    className="steps stepsClassName"
  >
    <li
      className="step stepClassName previous"
    >
      <i
        className="stepIcon stepIconClassName"
      >
        <Icon
          icon="checkmark"
          size="extra-small"
        />
      </i>
      <span
        className="stepTitle stepTitleClassName"
      >
        Select targets
      </span>
    </li>
    <li
      className="step stepClassName current"
    >
      <i
        className="stepIcon stepIconClassName"
      >
        2
      </i>
      <span
        className="stepTitle stepTitleClassName"
      >
        Configure setup
      </span>
    </li>
    <li
      className="step stepClassName"
    >
      <i
        className="stepIcon stepIconClassName"
      >
        3
      </i>
      <span
        className="stepTitle stepTitleClassName"
      >
        Schedule
      </span>
    </li>
  </ul>
</div>
`;
